In [ ]:
# Import necessary libraries

In [95]:
# import library
import pandas as pd
from chembl_webresource_client.new_client import new_client

In [125]:
# dir(new_client)
# 'activity',
#  'activity_supplementary_data_by_activity',
#  'assay',
#  'assay_class',
#  'atc_class',
#  'binding_site',
#  'biotherapeutic',
#  'cell_line',
#  'chembl_id_lookup',
#  'chembl_release',
#  'compound_record',
#  'compound_structural_alert',
#  'description',
#  'document',
#  'document_similarity',
#  'drug',
#  'drug_indication',
#  'drug_warning',
#  'go_slim',
#  'image',
#  'mechanism',
#  'metabolism',
#  'molecule',
#  'molecule_form',
#  'official',
#  'organism',
#  'protein_classification',
 # 'similarity',
 # 'source',
 # 'substructure',
 # 'target',
 # 'target_component',
 # 'target_relation',
 # 'tissue',
 # 'xref_source'

In [97]:
# Target search for coronavirus
target = new_client.target
target_query = target.search('aromatase')
targets = pd.DataFrame.from_dict(target_query)
targets
     

cross_references           organism  \
0  [{'xref_id': 'P11511', 'xref_name': None, 'xre...       Homo sapiens   
1  [{'xref_id': 'P22443', 'xref_name': None, 'xre...  Rattus norvegicus   

              pref_name  score  species_group_flag target_chembl_id  \
0  Cytochrome P450 19A1   20.0               False       CHEMBL1978   
1  Cytochrome P450 19A1   20.0               False       CHEMBL3859   

                                   target_components     target_type  tax_id  
0  [{'accession': 'P11511', 'component_descriptio...  SINGLE PROTEIN    9606  
1  [{'accession': 'P22443', 'component_descriptio...  SINGLE PROTEIN   10116

In [98]:
#search for targets in chembl database
target = new_client.target
target_query = target.search('aromatase')
targets = pd.DataFrame.from_dict(target_query)
targets

cross_references           organism  \
0  [{'xref_id': 'P11511', 'xref_name': None, 'xre...       Homo sapiens   
1  [{'xref_id': 'P22443', 'xref_name': None, 'xre...  Rattus norvegicus   

              pref_name  score  species_group_flag target_chembl_id  \
0  Cytochrome P450 19A1   20.0               False       CHEMBL1978   
1  Cytochrome P450 19A1   20.0               False       CHEMBL3859   

                                   target_components     target_type  tax_id  
0  [{'accession': 'P11511', 'component_descriptio...  SINGLE PROTEIN    9606  
1  [{'accession': 'P22443', 'component_descriptio...  SINGLE PROTEIN   10116

In [99]:
all_targets = target.all()
print(len(all_targets))

15598


In [100]:
# pd.DataFrame.from_dict(all_targets)

In [101]:
target = new_client.target
target_query_corona_virus = target.search('coronavirus')
targets_corona_virus = pd.DataFrame.from_dict(target_query_corona_virus)
targets_corona_virus

cross_references  \
0                                                 []   
1                                                 []   
2                                                 []   
3                                                 []   
4                                                 []   
5                                                 []   
6  [{'xref_id': 'P0C6U8', 'xref_name': None, 'xre...   
7                                                 []   
8  [{'xref_id': 'P0C6X7', 'xref_name': None, 'xre...   
9                                                 []   

                                            organism  \
0                                        Coronavirus   
1                                 Feline coronavirus   
2                                 Murine coronavirus   
3                                 Canine coronavirus   
4                             Human coronavirus 229E   
5                             Human coronavirus OC43   
6                                   SARS coronavirus   
7  Middle East respiratory syndrome-related coron...   
8                                   SARS coronavirus   
9    Severe acute respiratory syndrome coronavirus 2   

                                           pref_name  score  \
0                                        Coronavirus   17.0   
1                                 Feline coronavirus   14.0   
2                                 Murine coronavirus   14.0   
3                                 Canine coronavirus   14.0   
4                             Human coronavirus 229E   13.0   
5                             Human coronavirus OC43   13.0   
6                SARS coronavirus 3C-like proteinase   10.0   
7  Middle East respiratory syndrome-related coron...    9.0   
8                          Replicase polyprotein 1ab    4.0   
9                          Replicase polyprotein 1ab    4.0   

   species_group_flag target_chembl_id  \
0               False     CHEMBL613732   
1               False     CHEMBL612744   
2               False    CHEMBL5209664   
3               False    CHEMBL5291668   
4               False     CHEMBL613837   
5               False    CHEMBL5209665   
6               False       CHEMBL3927   
7               False    CHEMBL4296578   
8               False       CHEMBL5118   
9               False    CHEMBL4523582   

                                   target_components     target_type   tax_id  
0                                                 []        ORGANISM    11119  
1                                                 []        ORGANISM    12663  
2                                                 []        ORGANISM   694005  
3                                                 []        ORGANISM    11153  
4                                                 []        ORGANISM    11137  
5                                                 []        ORGANISM    31631  
6  [{'accession': 'P0C6U8', 'component_descriptio...  SINGLE PROTEIN   227859  
7                                                 []        ORGANISM  1335626  
8  [{'accession': 'P0C6X7', 'component_descriptio...  SINGLE PROTEIN   227859  
9  [{'accession': 'P0DTD1', 'component_descriptio...  SINGLE PROTEIN  2697049

In [102]:
targets_corona_virus.columns

Index(['cross_references', 'organism', 'pref_name', 'score',
       'species_group_flag', 'target_chembl_id', 'target_components',
       'target_type', 'tax_id'],
      dtype='object')

In [103]:
targets_corona_virus.cross_references[8]

[{'xref_id': 'P0C6X7', 'xref_name': None, 'xref_src': 'canSAR-Target'}]

* <u>**`target fields description`**</u>
* **cross_references:**

    - Description: Contains external references or links to the target in other databases or resources. These may include links to databases like
      UniProt, Ensembl, or other relevant biological databases.
    - Example: An array of dictionaries where each dictionary contains details about the external reference, such as the source and the external ID.

* organism:
    - Description: The scientific name of the organism from which the target is derived.
    - Example: "Homo sapiens" for human targets, "Mus musculus" for mouse targets.

* pref_name:
    - Description: The preferred name of the target. This is usually the most commonly used name for the target in scientific literature.
    - Example: "Beta-2 adrenergic receptor".

* score:
    - Description: This field typically represents a confidence score or relevance score associated with the target. The scoring system can vary and
     might indicate the reliability or significance of the target based on available data.
    - Example: 0.85 (Note: This field might not be present in all API responses).

* species_group_flag:
    - Description: A boolean flag indicating whether the target belongs to a species group, meaning it is a representative target for a group of     similar species.
    - Example: True or False.

* target_chembl_id:
    - Description: The unique identifier assigned to the target by ChEMBL. This ID is used to uniquely reference the target within the ChEMBL database.
    - Example: "CHEMBL210"

* target_components:
    - Description: Contains details about the components that make up the target. This is particularly relevant for complex targets like protein complexes or multi-subunit proteins.
    - Example: An array of dictionaries, where each dictionary includes information such as the component's description, accession numbers, and other relevant details.

* target_type:
    - Description: Describes the type or class of the target. Common target types include single proteins, protein complexes, protein families, and others.
    - Example: "SINGLE PROTEIN", "PROTEIN FAMILY".

* tax_id:
    - Description: The taxonomy ID of the organism from which the target is derived. This ID corresponds to the NCBI Taxonomy Database.
    - Example: 9606 for humans, 10090 for mice.
  
```json
{
  "cross_references": [
    {
      "xref_id": "P07550",
      "xref_name": "UniProtKB",
      "xref_src_db": "https://www.uniprot.org/uniprot/P07550"
    }
  ],
  "organism": "Homo sapiens",
  "pref_name": "Beta-2 adrenergic receptor",
  "score": 0.85,
  "species_group_flag": false,
  "target_chembl_id": "CHEMBL210",
  "target_components": [
    {
      "accession": "P07550",
      "component_description": "Beta-2 adrenergic receptor",
      "component_id": 10548,
      "component_type": "PROTEIN"
    }
  ],
  "target_type": "SINGLE PROTEIN",
  "tax_id": 9606
}
```
* <u>**`Detailed Field Usage`**</u>
    - cross_references: Useful for integrating ChEMBL data with other biological databases.
    - organism: Essential for understanding the biological context of the target.
    - pref_name: Helpful for identifying the target by its common name.
    - score: Useful for prioritizing targets based on confidence or relevance.
    - species_group_flag: Important for understanding whether the target represents a broader group.
    - target_chembl_id: Key identifier for referencing and querying ChEMBL data.
    - target_components: Crucial for detailed structural and functional insights, especially for complex targets.
    - target_type: Provides context about the nature of the target, aiding in categorization and analysis.
    - tax_id: Facilitates integration with taxonomic and other biological datasets.

<!-- <div style="background-color: #f0f0f0; padding: 10px; border-radius: 5px;">
<pre>
<code>
{
  "name": "John Doe",
  "age": 30,
  "city": "New York",
  "email": "johndoe@example.com"
}
</code>
</pre>
</div> -->

In [104]:
selected_target = targets_corona_virus.target_chembl_id[6]
selected_target

'CHEMBL3927'

In [105]:
# activity = new_client.activity
# res = activity.filter(target_chembl_id=selected_target)
#len(res) # ---->333

In [106]:
activity = new_client.activity
res = activity.filter(target_chembl_id=selected_target).filter(standard_type="IC50")
print(len(res))

133


In [107]:
len(res)

133

In [108]:
activity_coronavirus = pd.DataFrame.from_dict(res)

In [109]:
activity_coronavirus

action_type activity_comment  activity_id activity_properties  \
0          None             None      1480935                  []   
1          None             None      1480936                  []   
2          None             None      1481061                  []   
3          None             None      1481065                  []   
4          None             None      1481066                  []   
..          ...              ...          ...                 ...   
128        None             None     12041507                  []   
129        None             None     12041508                  []   
130        None             None     12041509                  []   
131        None             None     12041510                  []   
132        None             None     12041511                  []   

    assay_chembl_id                                  assay_description  \
0      CHEMBL829584  In vitro inhibitory concentration against SARS...   
1      CHEMBL829584  In vitro inhibitory concentration against SARS...   
2      CHEMBL830868  In vitro inhibitory concentration against SARS...   
3      CHEMBL829584  In vitro inhibitory concentration against SARS...   
4      CHEMBL829584  In vitro inhibitory concentration against SARS...   
..              ...                                                ...   
128   CHEMBL2150313  Inhibition of SARS-CoV PLpro expressed in Esch...   
129   CHEMBL2150313  Inhibition of SARS-CoV PLpro expressed in Esch...   
130   CHEMBL2150313  Inhibition of SARS-CoV PLpro expressed in Esch...   
131   CHEMBL2150313  Inhibition of SARS-CoV PLpro expressed in Esch...   
132   CHEMBL2150313  Inhibition of SARS-CoV PLpro expressed in Esch...   

    assay_type assay_variant_accession assay_variant_mutation bao_endpoint  \
0            B                    None                   None  BAO_0000190   
1            B                    None                   None  BAO_0000190   
2            B                    None                   None  BAO_0000190   
3            B                    None                   None  BAO_0000190   
4            B                    None                   None  BAO_0000190   
..         ...                     ...                    ...          ...   
128          B                    None                   None  BAO_0000190   
129          B                    None                   None  BAO_0000190   
130          B                    None                   None  BAO_0000190   
131          B                    None                   None  BAO_0000190   
132          B                    None                   None  BAO_0000190   

     ...   target_organism                     target_pref_name target_tax_id  \
0    ...  SARS coronavirus  SARS coronavirus 3C-like proteinase        227859   
1    ...  SARS coronavirus  SARS coronavirus 3C-like proteinase        227859   
2    ...  SARS coronavirus  SARS coronavirus 3C-like proteinase        227859   
3    ...  SARS coronavirus  SARS coronavirus 3C-like proteinase        227859   
4    ...  SARS coronavirus  SARS coronavirus 3C-like proteinase        227859   
..   ...               ...                                  ...           ...   
128  ...  SARS coronavirus  SARS coronavirus 3C-like proteinase        227859   
129  ...  SARS coronavirus  SARS coronavirus 3C-like proteinase        227859   
130  ...  SARS coronavirus  SARS coronavirus 3C-like proteinase        227859   
131  ...  SARS coronavirus  SARS coronavirus 3C-like proteinase        227859   
132  ...  SARS coronavirus  SARS coronavirus 3C-like proteinase        227859   

    text_value  toid  type units    uo_units upper_value  value  
0         None  None  IC50    uM  UO_0000065        None    7.2  
1         None  None  IC50    uM  UO_0000065        None    9.4  
2         None  None  IC50    uM  UO_0000065        None   13.5  
3         None  None  IC50    uM  UO_0000065        None  13.11  
4         None  None  IC50    uM  UO_0000065  

In [110]:
#activity_coronavirus.to_csv('bioactivity_data_coronavirus.csv', index=False)

In [111]:
bioactivity_data_coronavirus = pd.read_csv('bioactivity_data_coronavirus.csv')

In [112]:
print(len(bioactivity_data_coronavirus))
bioactivity_data_coronavirus.columns

133


Index(['action_type', 'activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type',
       'assay_variant_accession', 'assay_variant_mutation', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pref_name', 'target_tax_id', 'text_value', 'toid', 'type',
       'units', 'uo_units', 'upper_value', 'value'],
      dtype='object')

In [113]:
bioactivity_data_coronavirus = bioactivity_data_coronavirus[bioactivity_data_coronavirus.standard_value.notna()]

In [114]:
len(bioactivity_data_coronavirus)

133

In [115]:
bioactivity_class = []
for i in bioactivity_data_coronavirus.standard_value:
  if float(i) >= 10000:
    bioactivity_class.append("inactive")
  elif float(i) <= 1000:
    bioactivity_class.append("active")
  else:
    bioactivity_class.append("intermediate")

In [116]:
mol_cid = []
for i in bioactivity_data_coronavirus.molecule_chembl_id:
  mol_cid.append(i)

In [117]:
canonical_smiles = []
for i in bioactivity_data_coronavirus.canonical_smiles:
  canonical_smiles.append(i)

In [118]:
standard_value = []
for i in bioactivity_data_coronavirus.standard_value:
  standard_value.append(i)

In [119]:
data_tuples = list(zip(mol_cid, canonical_smiles, bioactivity_class, standard_value))
df3 = pd.DataFrame( data_tuples,  columns=['molecule_chembl_id', 'canonical_smiles', 'bioactivity_class', 'standard_value'])
df3.head()     

molecule_chembl_id                                 canonical_smiles  \
0       CHEMBL187579           Cc1noc(C)c1CN1C(=O)C(=O)c2cc(C#N)ccc21   
1       CHEMBL188487           O=C1C(=O)N(Cc2ccc(F)cc2Cl)c2ccc(I)cc21   
2       CHEMBL185698          O=C1C(=O)N(CC2COc3ccccc3O2)c2ccc(I)cc21   
3       CHEMBL426082              O=C1C(=O)N(Cc2cc3ccccc3s2)c2ccccc21   
4       CHEMBL187717  O=C1C(=O)N(Cc2cc3ccccc3s2)c2c1cccc2[N+](=O)[O-]   

  bioactivity_class  standard_value  
0      intermediate          7200.0  
1      intermediate          9400.0  
2          inactive         13500.0  
3          inactive         13110.0  
4      intermediate          2000.0

In [120]:
selection = ['molecule_chembl_id', 'canonical_smiles', 'standard_value']
df3 = bioactivity_data_coronavirus[selection]
df3

molecule_chembl_id                                   canonical_smiles  \
0         CHEMBL187579             Cc1noc(C)c1CN1C(=O)C(=O)c2cc(C#N)ccc21   
1         CHEMBL188487             O=C1C(=O)N(Cc2ccc(F)cc2Cl)c2ccc(I)cc21   
2         CHEMBL185698            O=C1C(=O)N(CC2COc3ccccc3O2)c2ccc(I)cc21   
3         CHEMBL426082                O=C1C(=O)N(Cc2cc3ccccc3s2)c2ccccc21   
4         CHEMBL187717    O=C1C(=O)N(Cc2cc3ccccc3s2)c2c1cccc2[N+](=O)[O-]   
..                 ...                                                ...   
128      CHEMBL2146517  COC(=O)[C@@]1(C)CCCc2c1ccc1c2C(=O)C(=O)c2c(C)c...   
129       CHEMBL187460        C[C@H]1COC2=C1C(=O)C(=O)c1c2ccc2c1CCCC2(C)C   
130       CHEMBL363535             Cc1coc2c1C(=O)C(=O)c1c-2ccc2c(C)cccc12   
131       CHEMBL227075      Cc1cccc2c3c(ccc12)C1=C(C(=O)C3=O)[C@@H](C)CO1   
132        CHEMBL45830             CC(C)C1=Cc2ccc3c(c2C(=O)C1=O)CCCC3(C)C   

     standard_value  
0            7200.0  
1            9400.0  
2           13500.0  
3           13110.0  
4            2000.0  
..              ...  
128         10600.0  
129         10100.0  
130         11500.0  
131         10700.0  
132         78900.0  

[133 rows x 3 columns]

In [121]:
df3 = pd.concat([df3,pd.Series(bioactivity_class)], axis=1)

In [122]:
df3

molecule_chembl_id                                   canonical_smiles  \
0         CHEMBL187579             Cc1noc(C)c1CN1C(=O)C(=O)c2cc(C#N)ccc21   
1         CHEMBL188487             O=C1C(=O)N(Cc2ccc(F)cc2Cl)c2ccc(I)cc21   
2         CHEMBL185698            O=C1C(=O)N(CC2COc3ccccc3O2)c2ccc(I)cc21   
3         CHEMBL426082                O=C1C(=O)N(Cc2cc3ccccc3s2)c2ccccc21   
4         CHEMBL187717    O=C1C(=O)N(Cc2cc3ccccc3s2)c2c1cccc2[N+](=O)[O-]   
..                 ...                                                ...   
128      CHEMBL2146517  COC(=O)[C@@]1(C)CCCc2c1ccc1c2C(=O)C(=O)c2c(C)c...   
129       CHEMBL187460        C[C@H]1COC2=C1C(=O)C(=O)c1c2ccc2c1CCCC2(C)C   
130       CHEMBL363535             Cc1coc2c1C(=O)C(=O)c1c-2ccc2c(C)cccc12   
131       CHEMBL227075      Cc1cccc2c3c(ccc12)C1=C(C(=O)C3=O)[C@@H](C)CO1   
132        CHEMBL45830             CC(C)C1=Cc2ccc3c(c2C(=O)C1=O)CCCC3(C)C   

     standard_value             0  
0            7200.0  intermediate  
1            9400.0  intermediate  
2           13500.0      inactive  
3           13110.0      inactive  
4            2000.0  intermediate  
..              ...           ...  
128         10600.0      inactive  
129         10100.0      inactive  
130         11500.0      inactive  
131         10700.0      inactive  
132         78900.0      inactive  

[133 rows x 4 columns]

In [123]:
# df3.to_csv('bioactivity_preprocessed_coronavirus.csv', index=False)

In [124]:
pd.read_csv('bioactivity_preprocessed_coronavirus.csv')

molecule_chembl_id                                   canonical_smiles  \
0         CHEMBL187579             Cc1noc(C)c1CN1C(=O)C(=O)c2cc(C#N)ccc21   
1         CHEMBL188487             O=C1C(=O)N(Cc2ccc(F)cc2Cl)c2ccc(I)cc21   
2         CHEMBL185698            O=C1C(=O)N(CC2COc3ccccc3O2)c2ccc(I)cc21   
3         CHEMBL426082                O=C1C(=O)N(Cc2cc3ccccc3s2)c2ccccc21   
4         CHEMBL187717    O=C1C(=O)N(Cc2cc3ccccc3s2)c2c1cccc2[N+](=O)[O-]   
..                 ...                                                ...   
128      CHEMBL2146517  COC(=O)[C@@]1(C)CCCc2c1ccc1c2C(=O)C(=O)c2c(C)c...   
129       CHEMBL187460        C[C@H]1COC2=C1C(=O)C(=O)c1c2ccc2c1CCCC2(C)C   
130       CHEMBL363535             Cc1coc2c1C(=O)C(=O)c1c-2ccc2c(C)cccc12   
131       CHEMBL227075      Cc1cccc2c3c(ccc12)C1=C(C(=O)C3=O)[C@@H](C)CO1   
132        CHEMBL45830             CC(C)C1=Cc2ccc3c(c2C(=O)C1=O)CCCC3(C)C   

     standard_value             0  
0            7200.0  intermediate  
1            9400.0  intermediate  
2           13500.0      inactive  
3           13110.0      inactive  
4            2000.0  intermediate  
..              ...           ...  
128         10600.0      inactive  
129         10100.0      inactive  
130         11500.0      inactive  
131         10700.0      inactive  
132         78900.0      inactive  

[133 rows x 4 columns]

In [11]:
import scanpy as sc
import pandas as pd
import GEOparse

In [13]:
geo_accession = "GSE130001"
gse = GEOparse.get_GEO_file(geo_accession, destdir="./")

29-Jul-2024 14:13:40 DEBUG utils - Directory ./ already exists. Skipping.
29-Jul-2024 14:13:40 INFO GEOparse - File already exist: using local version.


In [10]:
import scanpy as sc
import pandas as pd
import GEOparse

# Define the GEO accession number for the dataset you want to fetch
geo_accession = "GSE130001"  # Replace with your specific accession number

# Fetch the GEO dataset
gse = GEOparse.get_GEO(geo=geo_accession, destdir="./")
print(gse.gsms.items())
# Combine all samples into a single DataFrame
data_frames = []
for gsm_name, gsm in gse.gsms.items():
    print("-------",gsm_name, gsm)
    print(gsm.table)
    data_frames.append(gsm.table)
print(data_frames)
# Assuming the first column is gene identifiers and the rest are counts
expression_data = pd.concat(data_frames, axis=1)
expression_data = expression_data.set_index(expression_data.columns[0])

# Create an AnnData object
adata = sc.AnnData(expression_data.T)

# Perform standard preprocessing steps
sc.pp.normalize_total(adata, target_sum=1e4)
sc.pp.log1p(adata)
sc.pp.highly_variable_genes(adata)
adata = adata[:, adata.var.highly_variable]

# Scale the data and run PCA
sc.pp.scale(adata, max_value=10)
sc.tl.pca(adata, svd_solver='arpack')

# Run UMAP and clustering
sc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)
sc.tl.umap(adata)
sc.tl.leiden(adata)

# Visualize the data
sc.pl.umap(adata, color=['leiden'])

25-Jul-2024 18:31:36 DEBUG utils - Directory ./ already exists. Skipping.
25-Jul-2024 18:31:36 INFO GEOparse - File already exist: using local version.
25-Jul-2024 18:31:36 INFO GEOparse - Parsing ./GSE130001_family.soft.gz: 
25-Jul-2024 18:31:36 DEBUG GEOparse - DATABASE: GeoMiame
25-Jul-2024 18:31:36 DEBUG GEOparse - SERIES: GSE130001
25-Jul-2024 18:31:36 DEBUG GEOparse - PLATFORM: GPL16791
25-Jul-2024 18:31:36 DEBUG GEOparse - SAMPLE: GSM3729178
25-Jul-2024 18:31:36 DEBUG GEOparse - SAMPLE: GSM3729179


dict_items([('GSM3729178', <SAMPLE: GSM3729178>), ('GSM3729179', <SAMPLE: GSM3729179>)])
------- GSM3729178 <SAMPLE: GSM3729178>
Empty DataFrame
Columns: []
Index: []
------- GSM3729179 <SAMPLE: GSM3729179>
Empty DataFrame
Columns: []
Index: []
[Empty DataFrame
Columns: []
Index: [], Empty DataFrame
Columns: []
Index: []]


IndexError: index 0 is out of bounds for axis 0 with size 0

In [63]:
# drugs = new_client.drug

In [64]:
# len(drugs.all())

In [65]:
# molecule = new_client.molecule

In [66]:
# len(molecule.all())

In [67]:
# pd.DataFrame.from_dict(molecule.all()[:10]).molecule_structures[0]

In [68]:
# len(new_client.organism.all())